### PR TITLE
Enabling initial DocC documentation of Automerge-swift

### PR DESCRIPTION
**Pull request by heckj**
As I was starting in on using Automerge through swift, I thought I might contribute back some basic documentation. This PR has the working elements to enable DocC rendering of the documentation - right now just within Xcode (and any projects that use Automerge-swift as a dependency).

I replicated the content of the repositories README into an article within the documentation, made links to the relevant methods, and generally stubbed out a structure.

### CODE REVIEW

**Comment by lightsprint09** [comment]
This looks fantastic. Really looking forward to have proper DocC Support and even see how it works.

File: Package.resolved [hunk 1]
```diff
@@ -6,8 +6,8 @@
         "repositoryURL": "https://github.com/lightsprint09/automerge-swift-backend",
         "state": {
           "branch": null,
-          "revision": "5c1f3c58d078b4c2d64197129c30d2a7429aa44a",
-          "version": "0.1.6"
+          "revision": "3fa3f1e8b18eeb22a456d8bec813d2a266a6c57d",
```
Comment (lightsprint09) [inline Package.resolved]: Is this needed? Or just some leftover?

File: Package.swift [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:5.3
+// swift-tools-version:5.5
```
Comment (lightsprint09) [inline Package.swift:1]: Why is this required?

File: Package.swift [hunk 3]
```diff
@@ -40,7 +40,7 @@ let package = Package(
         )
     ],
     dependencies: [
-        .package(name: "AutomergeBackend", url: "https://github.com/lightsprint09/automerge-swift-backend", from: "0.1.6"),
+        .package(name: "AutomergeBackend", url: "https://github.com/lightsprint09/automerge-swift-backend", from: "0.1.9"),
```
Comment (lightsprint09) [inline Package.swift]: Looks like another leftover from a previous commit?

File: Package.resolved (hunk 1 above)
Comment (heckj) [reply Package.resolved]: Not required - just left-over from my pulling in the version you made and released the other day to bring this up to the latest backend implementation. I'll pull it so I'm not running into cross-concerns within a single PR.

File: Package.swift (hunk 2 above)
Comment (heckj) [reply Package.swift:1]: By setting the tools version to 5.5, this allows Xcode to know "automagically" that there could be a documentation archive included within the code, and uses that when it builds documentation. With an earlier version (such as 5.3 or even 5.4), the documentation archive files are ignored while building the documentation - effectively removing them from any organizational capacity, tossing any articles you may have included, etc.

If the project has an `.xcodeproj` for the library, and not just a `Package.swift` file, then you can build the documentation and use a documentation archive with that, but since this was a "pure swift library" setup, I didn't want to introduce an Xcodeproject wrapped _just_ to include the documentation.

File: Package.swift (hunk 3 above)
Comment (heckj) [reply Package.swift]: yep, will remove/reset

File: Tests/DocumentTest.swift [hunk 4]
```diff
@@ -231,7 +231,21 @@ class DocumentTest: XCTestCase {
                                 Op(action: .set, obj: .root, key: "now", insert: false, value: .float64(now.timeIntervalSince1970 * 1000), datatype: .timestamp, pred: [])
             ]))
         }
-
+    
+    // Code snippet in Document.change overview.
+    func testDocSnippetExample() {
```
Comment (lightsprint09) [inline Tests/DocumentTest.swift]: Could we move this into a separate file. Maybe DocumentSnippetsTest.swift?

File: Source/DataTypes/Text.swift [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
-//  File.swift
+//  Text.swift
 //  
 //
 //  Created by Lukas Schmidt on 21.04.20.
 //
 
 import Foundation
 
+/// A model that represents text.
```
Comment (lightsprint09) [inline Source/DataTypes/Text.swift]: would replace model with data structure

File: Source/DataTypes/Table.swift [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
-//  File.swift
+//  Table.swift
 //  
 //
 //  Created by Lukas Schmidt on 19.04.20.
 //
 
 import Foundation
 
+/// A model that represents a table of rows.
```
Comment (lightsprint09) [inline Source/DataTypes/Table.swift]: would replace model with data structure

File: Source/DataTypes/Counter.swift [hunk 7]
```diff
@@ -7,6 +7,7 @@
 
 import Foundation
 
+/// A model that represents a counter.
```
Comment (lightsprint09) [inline Source/DataTypes/Counter.swift]: would replace model with data structure

File: Source/DataTypes/Actor.swift [hunk 8]
```diff
@@ -7,6 +7,10 @@
 
 import Foundation
 
+/// A struct that represents a collaborator for making changes to a document.
```
Comment (lightsprint09) [inline Source/DataTypes/Actor.swift]: A struct that represents a collaborator for making changes to a document.
->
Represents a collaborator for making changes to a document.

File: Source/Automerge.docc/Automerge.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+- <doc:AutomergeBasics>
+
+### Tracking collaborative changes
+
+- ``Document``
+- ``Proxy``
+- ``Actor``
+- ``Patch``
+
+## Built in models
```
Comment (lightsprint09) [inline Source/Automerge.docc/Automerge.md]: models -> data structures / data types

**Comment by heckj** [comment]
I think this is about as far as I can take things for an initial cut - and would make most sense to remove and rebase after #49 and #46 have been merged. 

The coverage this provides based on the (experimental) doc coverage metrics from `docc`:

```
   --- Experimental coverage output enabled. ---
                | Abstract        | Curated         | Code Listing
Types           | 68% (17/25)     | 96% (24/25)     | 0.0% (0/25)
Members         | 11% (55/507)    | 82% (415/507)   | 0.20% (1/507)
Globals         | 5.7% (2/35)     | 97% (34/35)     | 0.0% (0/35)
```

I have a general intention of using the APIs more, working them end to end, and perhaps against another Automerge API endpoint (something I cobble as an experiment) - and from that I think I could create more "how to use" information.

The general pattern of the articles loosely follows the examples in the README, with some of that content having been integrated down into the relevant classes or structs. The notable lack is anything really detailing how to use a more "raw" document type of `Map`, `List`, `Text`, or `Counter`.

File: Source/Automerge.docc/AutomergeBasics.md [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+The following snippet creates a document named `doc1` that contains an empty list of cards:
+
+```swift
+var doc1 = Automerge.Document(Cards(cards: []))
+```
+
+### Updating your Model through the Automerge Document
+
+Treat your document an an immutable view of your current model's state. 
+Don't change it directly, instead call the ``Document/change(message:_:)`` method with a closure, and update your model's state within that closure.
```
Comment (lightsprint09) [inline Source/Automerge.docc/AutomergeBasics.md]: The compiler prevents changing it directly

File: Source/Proxy/Proxy+Table.swift [hunk 11]
```diff
@@ -9,13 +9,17 @@ import Foundation
 
 public extension Proxy {
 
+    /// Adds a row you provide to the table model in your document.
+    /// - Returns: <#description#>
```
Comment (lightsprint09) [inline Source/Proxy/Proxy+Table.swift]: Looks like something is missing?

File: Tests/DocumentationSnippetsTest.swift [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+        
+        // Initialize documents with known actor IDs.
+        var doc1 = Document(Coordinate(), actor: actor1)
+        // replicate, rather than create a whole new doc...
+        var doc2 = Document<Coordinate>(data: doc1.save(), actor: actor2)
+        // Creates a new, indepdendent doc:
+        //        var doc2 = Document(Coordinate(), actor: actor2)
+
+        // Set the values independently.
+        doc1.change { proxy in
```
Comment (lightsprint09) [inline Tests/DocumentationSnippetsTest.swift]: I would rather call the parameter doc or document instead of proxy. Proxy sounds to technical.

**Review by lightsprint09 — APPROVED** [review]

File: Source/Automerge.docc/AutomergeBasics.md (hunk 10 above)
Comment (heckj) [reply Source/Automerge.docc/AutomergeBasics.md]: reworded to remove that admonition from earlier content

File: Tests/DocumentationSnippetsTest.swift (hunk 12 above)
Comment (heckj) [reply Tests/DocumentationSnippetsTest.swift]: easily switched - back to `doc`
